api/v1/aerospikecluster_validating_webhook.go

```diff
@@ -1235,6 +1239,24 @@ func validateNamespaceConfig(
 	return nil
 }
 
+func validateMRTFields(nsConf map[string]interface{}) error {
+	mrtField := isMRTFieldSet(nsConf)
+	scEnabled := IsNSSCEnabled(nsConf)
+
+	if !scEnabled && mrtField {
+		return fmt.Errorf("MRT fields are not allowed in non-SC namespace %v", nsConf)
+	}
+
+	return nil
+}
+
+func isMRTFieldSet(nsConf map[string]interface{}) bool {
+	_, isMRTDurationSet := nsConf["mrt-duration"]
+	_, isDisableMRTWritesSet := nsConf["disable-mrt-writes"]
+
+	return isDisableMRTWritesSet || isMRTDurationSet
+}
+
```

Comment:
nit:
We can make it generic by making a list of MRT related fields and looping over them